tests/components/trend/conftest.py

```diff
@@ -0,0 +1,32 @@
+"""Fixtures for Trend integration tests."""
+from collections.abc import Awaitable, Callable
+from typing import Any
+
+import pytest
+
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+ComponentSetup = Callable[[dict[str, Any]], Awaitable[None]]
+
+
+@pytest.fixture
+async def setup_component(hass: HomeAssistant) -> ComponentSetup:
+    """Set up the trend component."""
+
+    async def _inner_setup(params: dict[str, Any]) -> None:
+        assert await async_setup_component(
+            hass,
+            "binary_sensor",
+            {
+                "binary_sensor": {
+                    "platform": "trend",
+                    "sensors": {
+                        "test_trend_sensor": params,
+                    },
+                }
+            },
+        )
+        await hass.async_block_till_done()
+
+    return _inner_setup
```

Comment:
Hi.

I am trying to understand what it brings to create a fixture that returns a function instead of creating a function and use it directly. Resulting code would be like:

The function:

```python
async def setup_component(hass: HomeAssistant, params: dict[str, Any]) -> None:
    assert await async_setup_component(
        hass,
        "binary_sensor",
        {
            "binary_sensor": {
                "platform": "trend",
                "sensors": {
                    "test_trend_sensor": params,
                },
            }
        },
    )
    await hass.async_block_till_done()
```

And call to that function would be:

```python
async def test_basic_trend(
    hass: HomeAssistant,
    states: list[str],
    inverted: bool,
    expected_state: str,
):
    """Test trend with a basic setup."""
    await setup_component(
        hass,
        {
            "entity_id": "sensor.test_state",
            "invert": inverted,
        },
    )
[...]
```

Could you clarify?

Reply:
Nothing :) It would be the same, I think that was a leftover. Thanks for the hint, will change this.